File: seeker/snippet/Task1.java
//date: 2026-01-09T17:16:38Z
//url: https://api.github.com/gists/f07db2dcdbf07070174855da4efd5d7f
//owner: https://api.github.com/users/Intyrrot14

// Task1.java
public class Task1 {
    public static void main(String[] args) {
        Student student = new Student("Tamirlan", 20, "S12345");
        Teacher teacher = new Teacher("Togzan", 45, "T98765");

        System.out.println("Student Info:");
        student.displayInfo();

        System.out.println("\nTeacher Info:");
        teacher.displayInfo();

        System.out.println("\nUsing Polymorphism:");
        Person[] people = { student, teacher };
        for (Person p : people) {
            p.displayInfo(); // dynamic dispatch
        }
    }
}

// Base class
class Person {
    private String name;
    private int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() { return name; }
    public int getAge() { return age; }

    public void displayInfo() {
        System.out.println("Name: " + name + ", Age: " + age);
    }
}

// Derived class
class Student extends Person {
    private String studentId;

    public Student(String name, int age, String studentId) {
        super(name, age);
        this.studentId = studentId;
    }

    @Override
    public void displayInfo() {
        System.out.println("Name: " + getName() + ", Age: " + getAge() + ", Student ID: " + studentId);
    }
}

// Derived class
class Teacher extends Person {
    private String employeeId;

    public Teacher(String name, int age, String employeeId) {
        super(name, age);
        this.employeeId = employeeId;
    }

    @Override
    public void displayInfo() {
        System.out.println("Name: " + getName() + ", Age: " + getAge() + ", Employee ID: " + employeeId);
    }
}

File: seeker/snippet/task_1.py
#date: 2025-08-13T16:40:41Z
#url: https://api.github.com/gists/c16e305a437ee7787571bd8dbb39ee8c
#owner: https://api.github.com/users/DanielIvanov19

class NumberSet:
    def __init__(self, n):
        if n > 64:
            raise ValueError("n should be ≤ 64")
        self.n = n
        self.bits = 0  # Initial empty

    def add(self, num):
        if 0 <= num <= self.n:
            self.bits |= (1 << num)

    def remove(self, num):
        if 0 <= num <= self.n:
            self.bits &= ~(1 << num)

    def contains(self, num):
        if 0 <= num <= self.n:
            return (self.bits >> num) & 1 == 1
        return False

    def __repr__(self):
        return "{" + ", ".join(str(i) for i in range(self.n + 1) if self.contains(i)) + "}"
